scripts/_common.sh

```diff
@@ -0,0 +1,68 @@
+#
+# Common variables
+#
+
+# Application version
+VERSION="1.0.36"
+
+# Remote URL to fetch application source tarball
+APPLICATION_SOURCE_URL="https://github.com/kanboard/kanboard/archive/v${VERSION}.tar.gz"
+
+#
+# Common helpers
+#
+
+# Download and extract application sources to the given directory
+# usage: extract_application_to DESTDIR
+extract_application() {
+  local DESTDIR=$1
+  rc_tarball="${DESTDIR}/application.tar.gz"
+  wget -q -O "$rc_tarball" "$APPLICATION_SOURCE_URL" \
+|| ynh_die "Unable to download application tarball"
+  tar xf "$rc_tarball" -C "$DESTDIR" --strip-components 1 \
+    || ynh_die "Unable to extract application tarball"
+  rm "$rc_tarball"
+}
+
+# Execute a command as another user
+# usage: exec_as USER COMMAND [ARG ...]
+exec_as() {
+  local USER=$1
+  shift 1
+
+  if [[ $USER = $(whoami) ]]; then
+    eval $@
+  else
+    # use sudo twice to be root and be allowed to use another user
+    sudo sudo -u "$USER" $@
+  fi
+}
+
```

Comment:
This is so over-engineered IMO. This complicate the script for exactly no gain, the whole script is run as "admin" user by YunoHost and this is used only to run thing as admin. Plus I don't think apps script should contain such generic helper, if one is really needed it should be in yunohost core.